File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()
    # 2중 for문 대신 이렇게 비교하기
    for i in range(len(phone_book)-1):
        # 슬라이싱으로 접두어 부분만 비교하기
        if phone_book[i] == phone_book[i+1][:len(phone_book[i])]:
            return False
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.Arrays;
class Solution {
    public boolean solution(String[] phone_book) {
        String[] sorted = Arrays.copyOf(phone_book, phone_book.length);
        Arrays.sort(sorted);
        for(int i = 0; i < sorted.length - 1; i++){
            if(sorted[i+1].startsWith(sorted[i])) return false;
        }
        return true;
    }
}